### PR TITLE
NTGR-620, NTGR-621 - ajax add/edit auto issuance

### PR DESCRIPTION
**Pull request by Ndibui**
This PR updates add and edit auto-issuance functionality to use Ajax and show toasts.

### CODE REVIEW

File: tests/includes/test-class-accredible-learndash-admin-action-handler.php [hunk 1]
```diff
@@ -34,23 +34,38 @@ public function test_add_auto_issuance() {
...
+				'success' => true,
+				'data'    => array(
+					'message'     => 'Saved auto issuance successfully.',
+					'redirectUrl' => esc_url( $redirect_url ),
+				),
 			)
 		);
+
+		try {
+			Accredible_Learndash_Admin_Action_Handler::add_auto_issuance(
```
Comment (kei178) [inline tests/includes/test-class-accredible-learndash-admin-action-handler.php:50]: I think we should also test the returning value if it doesn't raise an exception.

File: includes/templates/admin-auto-issuance-form.php [hunk 2]
```diff
@@ -69,14 +69,15 @@ class="button accredible-button-outline-natural accredible-button-large"><?php e
...
-			<form id="issuance-form" action="admin.php?page=accredible_learndash_admin_action&action=<?php echo esc_attr( $accredible_learndash_form_action ); ?>" method="post">
+			<form id="issuance-form">
 				<?php if ( 'add_auto_issuance' === $accredible_learndash_form_action ) { ?>
 					<?php wp_nonce_field( $accredible_learndash_form_action, '_mynonce' ); ?>
 				<?php } else { ?>
 					<?php wp_nonce_field( $accredible_learndash_form_action . $accredible_learndash_issuance->id, '_mynonce' ); ?>
 					<input type="hidden" name="id" value="<?php echo esc_attr( $accredible_learndash_issuance_id ); ?>">
 				<?php } ?>
 
+				<input type="hidden" name="call_action" value="<?php echo esc_attr( $accredible_learndash_form_action ); ?>">
```
Comment (kei178) [inline includes/templates/admin-auto-issuance-form.php:80]: Question - Should we change the value to `edit_auto_issuance` after successfully creating an auto issuance?

File: includes/models/class-accredible-learndash-model-auto-issuance.php [hunk 3]
```diff
@@ -64,15 +66,15 @@ public static function validate( $data, $id = null ) {
 			$kind    = array_key_exists( 'kind', $data ) ? $data['kind'] : null;
 
 			if ( ! self::is_kind_valid( $kind ) ) {
-				wp_die( 'ERROR: ' . esc_attr( $kind ) . ' is an invalid kind.' );
+				throw new Exception( esc_attr( $kind ) . ' is an invalid kind.' );
```
Comment (kei178) [inline includes/models/class-accredible-learndash-model-auto-issuance.php:69]: Could you also replace `wp_die` in the parent class `Accredible_Learndash_Model::validate`?

**Review by kei178 — CHANGES_REQUESTED** [review]
@Ndibui - I left a couple of comments on small things.

File: tests/includes/test-class-accredible-learndash-admin-action-handler.php (hunk 1 above)
Comment (Ndibui) [reply tests/includes/test-class-accredible-learndash-admin-action-handler.php]: Added missing assertion.

File: tests/includes/test-class-accredible-learndash-admin-action-handler.php [hunk 4]
```diff
@@ -128,23 +143,37 @@ public function test_update_auto_issuance() {
...
+				'success' => true,
+				'data'    => array(
+					'message'     => 'Saved auto issuance successfully.',
+					'redirectUrl' => esc_url( $redirect_url ),
+				),
 			)
 		);
 
+		try {
+			Accredible_Learndash_Admin_Action_Handler::edit_auto_issuance(
```
[the remark this answers is not in the document]
Comment (Ndibui) [reply tests/includes/test-class-accredible-learndash-admin-action-handler.php]: Added missing assertion.

File: includes/templates/admin-auto-issuance-form.php (hunk 2 above)
Comment (Ndibui) [reply includes/templates/admin-auto-issuance-form.php]: Updated.

File: includes/models/class-accredible-learndash-model-auto-issuance.php (hunk 3 above)
Comment (Ndibui) [reply includes/models/class-accredible-learndash-model-auto-issuance.php:69]: Updated `Accredible_Learndash_Model::validate` and related tests.

**Review by kei178 — APPROVED** [review]
LGTM

**Review by octaviorequena — APPROVED** [review]
Nice work @Ndibui !

**Pull request merged**